pubtools/pulplib/_impl/model/repository/file.py

```diff
@@ -20,3 +30,77 @@ class FileRepository(Repository):
     )
 
     mutable_urls = attr.ib(default=attr.Factory(lambda: ["PULP_MANIFEST"]), type=list)
+
+    def upload_file(self, file_obj, relative_url=None):
+        """Upload a file to this repository.
+
+        Args:
+            file_obj (str, file object)
+                If it's a string, then it's the path of file to upload.
+                Else, it ought to be a file-like object, see:
+                https://docs.python.org/3/glossary.html#term-file-object
+
+            relative_url (str)
+                Path that should be used in remote repository, can either
+                be a path to a directory or a path to a file, e.g:
+                - if relative_url is 'foo/bar' and file_obj has name 'f.txt',
```

Comment:
This isn't correct, is it?  How would the code know in this case that you wanted foo/bar to be interpreted as a directory and not as a file?  It's perfectly valid to name a file 'foo/bar'...

I think in this example, you wanted: `foo/bar/` ?  You should probably make it explicit that a trailing slash is the method by which the code decides what to do.

Reply:
yes. I wanted /foo/bar/ :|